packages/nextjs/src/server/clerkMiddleware.ts

```diff
@@ -76,7 +76,8 @@ interface ClerkMiddleware {
 const clerkMiddlewareRequestDataStore = new Map<'requestData', AuthenticateRequestOptions>();
 export const clerkMiddlewareRequestDataStorage = new AsyncLocalStorage<typeof clerkMiddlewareRequestDataStore>();
 
-export const clerkMiddleware: ClerkMiddleware = (...args: unknown[]): any => {
+// @ts-expect-error - TS be angry about the return type
+export const clerkMiddleware: ClerkMiddleware = async (...args: unknown[]): Promise<any> => {
```

Comment:
I'm not sure if this one can be async 😢 it's a factory function to create a middleware export: https://clerk.com/docs/references/nextjs/clerk-middleware

Might need to think a bit about how we want to handle the changes to `clerkClient()`.

In reality, the async case only applies when it's used in the context of a request.

Reply:
Aight, will give it some 🤔